audiodata.cpp

```diff
@@ -185,9 +185,13 @@ namespace KeyFinder {
 
     while (readAt < samples.end()) {
       double mean = 0.0;
-      if (shortcut && numSamplesRemaining >= factor) {
+      if (shortcut) {
         mean = *readAt;
-        std::advance(readAt, factor);
+        if (numSamplesRemaining >= factor)
+            std::advance(readAt, factor);
+        else
+            // Signal to the while loop that we are done
+            readAt = samples.end();
```

Comment:
Please brace the statements and remove the comment i.e.
if (numSamplesRemaining >= factor) {
    std::advance(readAt, factor);
} else {
    readAt = samples.end();
}